administrator/components/com_joomgallery/helpers/file.php

```diff
@@ -367,301 +383,409 @@ public static function resizeImage(&$debugoutput, $src_file, $dest_file, $usefor
+          default:
+            $offsetx = floor(($srcWidth - ($new_width * $ratio)) * 0.5);
+            $offsety = floor(($srcHeight - ($new_height * $ratio)) * 0.5);
+            break;
+        }
       }
-    }
-
-    if(is_null($offsetx) && is_null($offsety))
-    {
-      $ratio = max($ratio, 1.0);
+      if(is_null($offsetx) && is_null($offsety))
+      {
+        $ratio = max($ratio, 1.0);
 
-      $destWidth  = (int)($srcWidth / $ratio);
-      $destHeight = (int)($srcHeight / $ratio);
-    }
-    else
-    {
-      $destWidth = $new_width;
-      $destHeight = $thumbheight;
-      $srcWidth  = (int)($destWidth * $ratio);
-      $srcHeight = (int)($destHeight * $ratio);
-    }
+        $destWidth  = (int)($srcWidth / $ratio);
+        $destHeight = (int)($srcHeight / $ratio);
+      }
+      else
+      {
+        $destWidth = $new_width;
+        $destHeight = $new_height;
+        $srcWidth  = (int)($destWidth * $ratio);
+        $srcHeight = (int)($destHeight * $ratio);
+      }
 
     // Method for creation of the resized image
     switch($method)
     {
       case 'gd1':
+        $debugoutput.='GD1...<br/>';
+        // no animated gif support
         if(!function_exists('imagecreatefromjpeg'))
+        // check, if GD is available
         {
           $debugoutput.=JText::_('COM_JOOMGALLERY_UPLOAD_GD_LIBARY_NOT_INSTALLED');
           return false;
         }
-        if($imginfo[2] == 'JPG')
+        // create empty image of specified size
+        $dst_frames = array(imagecreate($destWidth, $destHeight));
+        $src_frames = JoomFile::imageCreateFrom_GD($src_file, $dst_frames, $imginfo);
+
+        if (in_array(false, $src_frames))
         {
-          $src_img = imagecreatefromjpeg($src_file);
+          $debugoutput.=JText::_('COM_JOOMGALLERY_UPLOAD_GD_LIBARY_NOT_ABLE_RESIZING');
+          return false;
         }
-        else
+        if($angle > 0)
+        // rotate image, if needed
         {
-          if ($imginfo[2] == 'PNG')
+          foreach ($src_frames as $key => $frame)
           {
-            $src_img = imagecreatefrompng($src_file);
+            $src_frames[$key]['image'] = imagerotate($src_frames[$key]['image'], $angle, 0);
           }
-          else
-          {
-            $src_img = imagecreatefromgif($src_file);
-          }
-        }
-        if(!$src_img)
-        {
-          return false;
         }
 
-        if($angle > 0)
+        foreach ($src_frames as $key => $frame)
         {
-          $src_img = imagerotate($src_img, $angle, 0);
+          if (!is_null($offsetx) && !is_null($offsety))
+          // resizing with GD1
+          {
+            imagecopyresized( $dst_frames[$key], $src_frames[$key]['image'], 0, 0, $offsetx, $offsety,
+                              $destWidth, (int)$destHeight, $srcWidth, $srcHeight);
+          }
+          else
+          {
+            imagecopyresized( $dst_frames[$key], $src_frames[$key]['image'], 0, 0, 0, 0, $destWidth,
```

Comment:
resizing of the GD objects same as before. Only difference are the frames array (dst_frames,src_frames) instead of one single GD object (dst_img,src_img)